foundation-models/openai/src/main/java/com/sap/ai/sdk/foundationmodels/openai/StreamConverter.java

```diff
@@ -0,0 +1,108 @@
+package com.sap.ai.sdk.foundationmodels.openai;
+
+import static java.nio.charset.StandardCharsets.UTF_8;
+import static java.util.Spliterator.NONNULL;
+import static java.util.Spliterator.ORDERED;
+
+import io.vavr.control.Try;
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.io.UncheckedIOException;
+import java.util.Iterator;
+import java.util.NoSuchElementException;
+import java.util.Spliterators;
+import java.util.stream.Stream;
+import java.util.stream.StreamSupport;
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.hc.core5.http.HttpEntity;
+
+@Slf4j
+class StreamConverter {
+
+  /** see {@link BufferedReader#DEFAULT_CHAR_BUFFER_SIZE} */
+  static final int BUFFER_SIZE = 8192;
+
+  @FunctionalInterface
+  private interface ReadHandler<T> {
```

Comment:
do we really need this? Maybe we can use `Consumer<T>` instead?

Reply:
Consumer does not allow for checked exception.

I considered alternatives, e.g. VAVR exception types. But I appreciated the narrow declaration of thrown exceptions, i.e. only `IOException` in our case.

However, this means if `#readEntry` throws any unchecked exception the stream will not be closed via close-handler. 

Not a strong opinion. I may have to reconsider.

Reply:
I changed the code to avoid unnecessary nested types